### PR TITLE
run the tests against the http server & against the proxy

### PR DESCRIPTION
**Pull request by yanns**

### CODE REVIEW

File: src/proxy.rs [hunk 1]
```diff
@@ -180,14 +181,17 @@ impl Service for Proxy {
...
+pub fn listen(addr: SocketAddr, pool: Pool) -> result::Result<(thread::JoinHandle<()>, u16), io::Error> {
+    let (tx, rx) = channel();
     let handle = thread::spawn(move || {
 
         let mut core = Core::new().unwrap();
         let handle = core.handle();
 
         let listener = TcpListener::bind(&addr, &handle).unwrap();
-        info!("Listening on http://{}", &addr);
+        let local_addr = listener.local_addr().unwrap();
```
Comment (hjr3) [inline src/proxy.rs]: Should replace these calls to `unwrap` with `try!` or `?`.

Comment (hjr3) [inline src/proxy.rs]: Also, should we send back `local_addr` so we can get both IP and port that we are listening on?

Comment (yanns) [inline src/proxy.rs]: the code is inside the `thread::spawn`. Difficult to use a `try!` here.
Do you have an idea?

Comment (yanns) [inline src/proxy.rs]: I return now the `SocketAddr` instead of only the port.

Comment (hjr3) [inline src/proxy.rs]: gah you are right. now i know what there are other unwraps there!

**Pull request merged**